### PR TITLE
Say from which operator the asserted Context was taken

### PR DESCRIPTION
**Pull request by bsideup**

### CODE REVIEW

File: reactor-test/src/test/java/reactor/test/DefaultContextExpectationsTest.java [hunk 1]
```diff
@@ -292,7 +321,27 @@ public void notMatchesWithDescriptionAndScenarioName() {
...
+				.withMessageStartingWith("" +
+						"[scenario] Context doesn't match predicate someDescription\n" +
+						"Context: Context1{foo=bar}"
+				);
+	}
+
+	@Test
+	public void capturedOperator() {
+		assertContextExpectationFails(
+				s -> s.doOnEach(__ -> {}),
```
Comment (simonbasle) [inline reactor-test/src/test/java/reactor/test/DefaultContextExpectationsTest.java:333]: would it make sense / demonstrate additional value to make the transformation / tested chain the same in both `capturedOperator` and `capturedOperatorWithDebug`?

File: reactor-test/src/main/java/reactor/test/DefaultStepVerifierBuilder.java [hunk 2]
```diff
@@ -2525,7 +2525,7 @@ static String formatFusionMode(int m) {
 				Scannable lowest = Scannable.from(s);
 
 				//attempt to go back to the leftmost parent to check the Context from its perspective
-				Context c = Flux.<Scannable>
+				CoreSubscriber<?> subscriber = Flux.<Scannable>
```
Comment (simonbasle) [inline reactor-test/src/main/java/reactor/test/DefaultStepVerifierBuilder.java:2528]: wait, do we still need that? now the `subscriber` could be `null` and none of the `Consumer<CoreSubscriber>` have been amended to fix that :(

**Review by simonbasle — CHANGES_REQUESTED** [review]

File: reactor-test/src/main/java/reactor/test/DefaultStepVerifierBuilder.java (hunk 2 above)
Comment (bsideup) [reply reactor-test/src/main/java/reactor/test/DefaultStepVerifierBuilder.java:2528]: We do. Also, there is a check that verifies that the context is present before running any of the context assertions

Comment (simonbasle) [reply reactor-test/src/main/java/reactor/test/DefaultStepVerifierBuilder.java:2528]: yeah just re-checked the code and tests, this should be equivalent, as previously this resulting in a `Context` variable that was also `null`. My bad.

**Review by simonbasle — APPROVED** [review]

**Pull request merged**